src/proto/h1/dispatch.rs

```diff
@@ -31,7 +31,6 @@ pub(crate) trait Dispatch {
         cx: &mut task::Context<'_>,
     ) -> Poll<Option<Result<(Self::PollItem, Self::PollBody), Self::PollError>>>;
     fn recv_msg(&mut self, msg: crate::Result<(Self::RecvItem, Body)>) -> crate::Result<()>;
-    fn poll_ready(&mut self, cx: &mut task::Context<'_>) -> Poll<Result<(), ()>>;
```

Comment:
We should leave this in, it still have use in detecting that on the client side, the user no longer cares. The server should still say it's not ready if there is already a request `in_flight`.

Reply:
Same as below, this would also mean we have to leave `Service::poll_ready` for `Dispatch` implementation on `Server`.

Reply:
No, this could just use the "simple" version for a server, as described in the other comment.